k8s/src/main/scala/io/buoyant/k8s/EndpointsNamer.scala

```diff
@@ -55,13 +57,11 @@ class EndpointsNamer(idPrefix: Path, mkApi: String => v1.NsApi) extends Namer {
     def get(name: String): NsCache = synchronized {
       activity.sample() match {
         case Activity.Ok(snap) =>
-          snap.get(name) match {
-            case Some(ns) => ns
-            case None =>
-              val ns = mkNs(name)
-              activity() = Activity.Ok(snap + (name -> ns))
-              ns
-          }
+          snap.getOrElse(name, {
+            val ns = mkNs(name)
+            activity() = Activity.Ok(snap + (name -> ns))
+            ns
+          })
```

Comment:
I think this is harder to reason about, since we have no guarantees that this closure is called within the context of the synchronized block